### PR TITLE
Playlist fixes

### PR DESCRIPTION
**Pull request by chethus**
Some fixes and improvements to playlists
- Fixed server side error handling for nonexistent playlists
- Added handling for privated videos
Screenshot 2020-07-30 at 7 12 35 PM
- Improve thumbnail quality for all videos (select the best possible)
- Improve css so that it is clear how titles and thumbnails correspond

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Video.java [hunk 1]
```diff
@@ -71,8 +73,14 @@ public static Video videoFromPlaylistItem(PlaylistItem playlistItem) throws Malf
         String videoId = snippet.getResourceId().getVideoId();
         String videoTitle = snippet.getTitle();
         String videoDescription = snippet.getDescription();
-        String thumbnailUrl = snippet.getThumbnails().getDefault().getUrl();
         boolean isPublic = !playlistItem.getStatus().getPrivacyStatus().equals("private");
+        String thumbnailUrl;
+        if (!isPublic) {
+            // Set url to unavailable thumbnail.
+            thumbnailUrl = "http://i.ytimg.com/vi/D5_RLA5NJAY/hqdefault.jpg";
```
Comment (nikhilbirmiwal) [inline portfolio/src/main/java/com/google/sps/data/Video.java]: Let's define this URL as a variable somewhere. 

UNAVAILABLE_THUMBNAIL_URL = <>.

File: portfolio/src/main/java/com/google/sps/data/Video.java [hunk 2]
```diff
@@ -125,4 +133,20 @@ public void setIsPublic(boolean isPublic) {
...
+    public static String maxResThumbnailUrl(ThumbnailDetails details) {
+            Thumbnail[] thumbnails = new Thumbnail[]{
+                details.getMaxres(),
+                details.getStandard(),
+                details.getHigh(),
+                details.getMedium(),
+                details.getDefault()
+            };
+            Thumbnail highestRes = null;
+            for (int thumbnailInd = 0; highestRes == null; thumbnailInd ++) {
```
Comment (nikhilbirmiwal) [inline portfolio/src/main/java/com/google/sps/data/Video.java]: Please leave a comment indicating that these thumbnails are sorted.

**Review by nikhilbirmiwal — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/data/Video.java (hunk 1 above)
Comment (chethus) [reply portfolio/src/main/java/com/google/sps/data/Video.java]: Fixed, added as static variable

File: portfolio/src/main/java/com/google/sps/data/Video.java (hunk 2 above)
Comment (chethus) [reply portfolio/src/main/java/com/google/sps/data/Video.java]: Added.

**Pull request merged**